### ISSUE TITLE
Edit API (1/4): Client Secret

### ISSUE THREAD

**Issue by Zoefinders** [PrestaShop/PrestaShop#33718]
### Description
As a shop admin
I want to be able create or regenerate client secret 
So to have a secure client secret

#### Business rules

- Client secret will be visible only once 
- Preview client secret will be automatically revoke when the demand for regenerated is lanched 
- The client secret must copy thanks to "copy" button (Nice to have) 

#### Acceptance tests
**Acceptance test 1:API acess _creation_**

**Given** i am in "API access List " page
**When** 
1. I click on " Add new API access & "Edit %API access name% page open
2.  At a minimum I must fill : 
- "Client ID" filed 
-"Client name" filed
3. I click on " generate client secret & save" 

**Then** 
The Api acces is created. 
"API access list" page open with two banner 
1. Bleu banner with confirmation message & warning for de client secret code 
_The API access and Client secret has been generated successfully. 
Note that client secret will be visbile only once and must be saved_ 

2. Green banner with the clients secret code and copy button
Exemple client secret` 8765647TYRE4T 

For close the banner i need refresh the page 

**Acceptance test 1:API acess _edtion_**
**Given** i am in "API access List " page
**When** 
1. I click on one API access in the list & "Edit %API access name% page open
2. I click on "Regeneate client secret" 

**Then** 
"API access list" page open with two banner 
1. Bleu banner with confirmation message & warning for de client secret code 
_Client secret has been generated successfully. Your former secret client is now an obselette
Note that client secret will be visbile only once and must be saved_ 

2. Green banner with the clients secret code and copy button
Exemple client secret `8765647TYRE4T` 

For close the banner i need refresh the page 

### QA Scenarios
* Functional > BO > Advanced Parameters > Authorization Server > Add API Access
Update 

|**Action** | **Additional** | **Expected** |
|-----------|-----------|-----------|
| | **Main Test** | |
Click on the button "Add new API access"|   | New API access Page is displayed correctly
Fill the form with following data (Client ID : "Client XYZ", Client Name: "ClientName ABC")<br>Click on "Generate client secret & Save" button| | There is 1 application in the list<br>The message “The API access and Client secret has been generated sucessfully” is displayed<br>The message "Client secret: " is available<br>The button Copy put the client secret in the clipboard

to 

|**Action** | **Additional** | **Expected** |
|-----------|-----------|-----------|
| | **Main Test** | |
Click on the button "Add new API access"|   | New API access Page is displayed correctly
Fill the form with following data (Client ID : "Client XYZ", Client Name: "ClientName ABC")<br>Click on "Generate client secret & Save" button| | There is 1 application in the list<br>The message “The API access and Client secret has been generated sucessfully” is displayed<br>The message "Client secret: " is available<br>The button Copy put the client secret in the clipboard
Reload the page | | There is 1 application in the list<br>The message “The API access and Client secret has been generated sucessfully” is displayed<br>The message "Client secret: " is not available

* Functional > BO > Advanced Parameters > Authorization Server > Edit API Access
Update

|**Action** | **Additional** | **Expected** |
|-----------|-----------|-----------|
| | **Main Test** | |
Fill the form with following data (Client ID : "Client UVW", Client Name: "ClientName DEF")<br>Click on "Save" button| | There is 1 application in the list<br>The message “The API access and Client secret has been updated sucessfully” is displayed

to 

|**Action** | **Additional** | **Expected** |
|-----------|-----------|-----------|
| | **Main Test** | |
Fill the form with following data (Client ID : "Client UVW", Client Name: "ClientName DEF")<br>Click on "Save" button| | There is 1 application in the list<br>The message “The API access and Client secret has been updated sucessfully” is displayed
On the first row, click on the edit button | | Edit API access Page is displayed correctly
Click on the button "Regenerate a client secret" | | There is 1 application in the list<br>The message “_Client secret has been generated successfully. Your former secret client is now an obselette  Note that client secret will be visbile only once and must be saved_” is displayed<br>The message "Client secret: " is available<br>The button Copy put the client secret in the clipboard

### Additional context

[Figma link](https://www.figma.com/file/uaRxD74AMc98kLlyyhy0RM/Authorization-server-management?type=design&node-id=1041-655&mode=design&t=rU5kcuGFOARrcKYs-0)

**Comment by jolelievre** [comment]
Interesting https://www.oauth.com/oauth2-servers/client-registration/client-id-secret/

**Comment by MatShir** [comment]
blocked by https://github.com/PrestaShop/PrestaShop/issues/33765
